test/e2e/helpers.js

```diff
@@ -203,16 +203,20 @@ async function withFixtures(options, testSuite) {
       }
       const contracts =
         smartContract instanceof Array ? smartContract : [smartContract];
-      await Promise.all(
-        contracts.map((contract) => seeder.deploySmartContract(contract)),
-      );
+
+      const hardfork =
+        localNodeOptsNormalized[0].options.hardfork || 'muirGlacier';
+      for (const contract of contracts) {
+        await seeder.deploySmartContract(contract, hardfork);
+      }
+
```

Comment:
we need to pass the hardfork used, as with viem the default writeContract method uses type 2 tx's, so we need a way to indicate to use legacy when we need it